code/controller/controller.ino

```diff
@@ -148,31 +141,27 @@ void reset() {
 }
 
 #ifdef ENABLE_SERIAL
-void printAnalog(const long& packedVal) {
+void printAnalog(int16_t xval, int16_t yval) {
 
-  int tx = packedVal >> (sizeof(int) * 8);
-  int ty = packedVal & 0xFFFFL;
-  sprintf(msg, "Raw: %lx, x-axis: %d, y-axis: %d", packedVal, tx, ty);
+  snprintf(msg, MSG_LEN-1, "x-axis: %d, y-axis: %d", xval, yval);
   Serial.println(msg);
 }
 #else
-inline void printAnalog(const long& packedVal) {}
+inline void printAnalog(const int16_t xval, const int16_t yval) {}  // shouldn't need this at all, because all printAnalog() should be wrapped; just in case.
 #endif
 
-unsigned long readAnalog(int xPin, int yPin, const int XMIN, const int XMAX, const int YMIN, const int YMAX, bool returnRaw = false) {
-  int readX = analogRead(xPin); // store in temporary variables to use in constrain()
-  int readY = analogRead(yPin); 
+int16_t readAnalog(int pin, const int MIN, const int MAX, bool returnRaw=false) {
```

Comment:
This is much better than trying to return both values packed together. Small nitpick: in this case, I'd prefer for it to be called "readAxis". Also no need to capitalize MIN and MAX.

Reply:
yeah.  i kept as much of your old stuff as possible.  i was also confused as to why you were packing and then immediately unpacking.

Reply:
Haha, I was refactoring it late at night and I wanted a single function call to return both x and y axes. Wasn't worth it.